### ISSUE TITLE
Check if the editor can be included with the next release

### ISSUE THREAD

**Issue by hiker** [supertuxkart/stk-code#1581]
Since the editor uses stk-assets, it won't be a significant bigger download (approx. 5 MB in an additional assets directory, 3 MB in icons etc, plus code and library). But it would expose a lot more users to the editor.

**Comment by hiker** [comment]
It's included with windows now, it needs to be checked for osx ( @auriamg ) and linux ( @deveee ).

**Comment by deveee** [comment]
@hiker I downloaded mcsab fork of irrlicht, patched it make it working with mesa 10 and I was able to compile stk-editor.

But I still can't run it. I'm not sure if I'm doing something wrong or that's a bug in code because in README I see only few words about "editor" directory and anything more. At this moment I get segfault on startup.

I created a ticket:
https://github.com/mcsab/stk-editor/issues/19

**Comment by hiker** [comment]
Added, and write permission given ;)

**Comment by hiker** [comment]
@deveee or @mcsab could you update our with the latest changes for the editor?
@auriamg what about osx?

**Comment by deveee** [comment]
I fixed crash on startup and did some work on cmake support. Now irrlicht
is statically linked in the same way as in STK. I just need to add physfs
library to package and it should work.

osx needs some work in cmake files because I don't have computer where I
could check it.

**Comment by deveee** [comment]
Recently we dropped support for old tracks. Stk-editor creates tracks in version 5. If we want to include editor in upcoming STK version, we should allow to run tracks which were created in editor.

As a very ugly workaround, we can bump stk-editor exporter to version 6. Tracks created in editor still look good enough.

Btw. in stk-assets repo we don't have editor/maps directory. It is necessary to save files. I fixed it in this commit: 7e7aa9a86b1aacf4a340f9554847e2cea87f03b6
